modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/timer/test/TimerExecutorTest.java

```diff
@@ -264,25 +276,27 @@ private KaleoTaskInstanceToken _addKaleoTaskInstanceToken(
 			Collections.emptyList(), null, _workflowContext, _serviceContext);
 	}
 
-	private void _executeTimer(
-			KaleoInstanceToken kaleoInstanceToken,
-			KaleoTaskInstanceToken kaleoTaskInstanceToken,
-			KaleoTimerInstanceToken kaleoTimerInstanceToken)
+	private void _executeTimer(KaleoTimerInstanceToken kaleoTimerInstanceToken)
 		throws Exception {
 
-		Method executeTimerMethod = ReflectionUtil.getDeclaredMethod(
-			_timerExecutor.getClass(), "_executeTimer", ExecutionContext.class);
+		String schedulerGroupName = SchedulerUtil.getGroupName(
+			kaleoTimerInstanceToken.getCompanyId(),
+			kaleoTimerInstanceToken.getKaleoTimerInstanceTokenId());
+
+		SchedulerResponse schedulerResponse =
+			SchedulerEngineHelperUtil.getScheduledJob(
+				schedulerGroupName, schedulerGroupName, StorageType.PERSISTED);
 
-		executeTimerMethod.invoke(
-			_timerExecutor,
-			new ExecutionContext(
-				kaleoInstanceToken, kaleoTimerInstanceToken, _workflowContext,
-				_serviceContext) {
+		Message message = schedulerResponse.getMessage();
 
-				{
-					setKaleoTaskInstanceToken(kaleoTaskInstanceToken);
-				}
-			});
+		message.put(
+			SchedulerEngine.DESTINATION_NAME,
+			KaleoRuntimeDestinationNames.WORKFLOW_TIMER);
+		message.put(SchedulerEngine.GROUP_NAME, schedulerGroupName);
+		message.put(SchedulerEngine.JOB_NAME, schedulerGroupName);
+		message.put("companyId", kaleoTimerInstanceToken.getCompanyId());
```

Comment:
@pedro-oliveira446 `companyId` message should comes before `DESTINATION_NAME`, sort it please.

Thank you!